template/sink.c.erb

```diff
@@ -59,11 +59,27 @@ sink_consume(bt_self_component_sink *self_component_sink) {
     const bt_event_class *event_class = bt_event_borrow_class_const(event);
 
     /* Call dispatcher */
-    name_to_dispatcher_t *s = NULL;
+    bool is_callback_called = false;
     const char *class_name = bt_event_class_get_name(event_class);
-    HASH_FIND_STR(common_data->name_to_dispatcher, class_name, s);
-    if (s)
-      (*((dispatcher_t(*))(s->dispatcher)))(s->callbacks, common_data, event);
+
+    {
+      name_to_dispatcher_t *s = NULL;
+      HASH_FIND_STR(common_data->name_to_dispatcher, class_name, s);
+      if (s) {
+        (*((dispatcher_t(*))(s->dispatcher)))(s->callbacks, common_data, event, &is_callback_called);
+      }
+    }
+
+    {
+      const bt_stream_class* stream_class = bt_event_class_borrow_stream_class_const(event_class);
+      const char* stream_class_name = bt_stream_class_get_name(stream_class);
+
+      name_to_dispatcher_t *s = NULL;
+      HASH_FIND_STR(common_data->name_to_matching_dispatcher, stream_class_name, s);
+      if (s) {
+        (*((dispatcher_t(*))(s->dispatcher)))(s->callbacks, common_data, event, &is_callback_called);
```

Comment:
Let's put a comment to say that we know `is_callback_called` will only be modified if a callback is present.